### PR TITLE
feat: add functions for checking robot state

### PR DESCRIPTION
**Pull request by doinkythederp**
Mirrors c++ api, more or less

### CODE REVIEW

File: pros-sys/src/misc.rs [hunk 1]
```diff
@@ -2,11 +2,14 @@ use core::ffi::*;
 
 pub const NUM_V5_PORTS: usize = 21;
 // v5 comp
-pub const COMPETITION_DISABLED: i32 = 0b001;
```
Comment (Gavin-Niederman) [inline pros-sys/src/misc.rs:5]: are you sure that these are supposed to be u8?

**Review by Gavin-Niederman — COMMENTED** [review]
Just a simple question before merging

File: pros-sys/src/misc.rs (hunk 1 above)
Comment (doinkythederp) [reply pros-sys/src/misc.rs:5]: `competition_get_state` returns a u8 bitfield so presumably its members are u8's

Comment (doinkythederp) [reply pros-sys/src/misc.rs:5]: See: https://pros.cs.purdue.edu/v5/api/c/misc.html#competition-get-status

Comment (Gavin-Niederman) [reply pros-sys/src/misc.rs:5]: ok, cool.

**Review by Gavin-Niederman — APPROVED** [review]
LGTM (Lets Go To Mordor)

**Pull request merged**